src/pages/ConnectWallet/MobileConnect.tsx

```diff
@@ -91,6 +91,17 @@ export const MobileConnect = () => {
   }, [setDefaultRoute, onOpen])
 
   const query = useQuery<{ returnUrl: string }>()
+
+  useEffect(() => {
+    if (state.isLoadingLocalWallet) {
+      setIsWaitingForRedirection(true)
+    }
+
+    if (!state.isLoadingLocalWallet && !hasWallet) {
+      setIsWaitingForRedirection(false)
+    }
+  }, [state.isLoadingLocalWallet, hasWallet])
```

Comment:
Quite hacky but works very well without thouchingthe WalletProvider monstruosity and keeping the state local to this particular scope

If the local wallet is loading, we are going to wait for the redirection, if the local wallet isn't loading and we don't have access to the wallet, we assume the user cancelled the biometric and so we are displaying the wallet list etc